runtime/command-metrics/src/main/java/io/aklivity/zilla/runtime/command/metrics/internal/airline/ZillaMetricsCommand.java

```diff
@@ -73,44 +61,47 @@ public final class ZillaMetricsCommand extends ZillaCommand
     @Arguments(title = {"name"})
     public List<String> args;
 
-    public ZillaMetricsCommand()
-    {
-        Path engineDirectory = engineDirectory();
-        this.labelsPath = engineDirectory;
-        this.metricsPath = engineDirectory.resolve("metrics");
-    }
-
     @Override
     public void run()
     {
         String binding = args != null && args.size() >= 1 ? args.get(0) : null;
-        Map<Metric.Kind, List<MetricsLayout>> layouts = Map.of();
-        try
+        Engine engine = engine();
+        requireNonNull(engine);
+        MetricsReader metrics = new MetricsReader(engine, engine::supplyLocalName);
+        LongPredicate filterPredicate = filterPredicate(namespace, binding, engine::supplyLabelId);
+        List<MetricRecord> records = filter(metrics.records(), filterPredicate);
+        MetricsPrinter printer = new MetricsPrinter(records);
+        do
         {
-            layouts = Map.of(
-                    COUNTER, countersLayouts(),
-                    GAUGE, gaugesLayouts(),
-                    HISTOGRAM, histogramsLayouts());
-            final LabelManager labels = new LabelManager(labelsPath);
-            MetricsProcessor metrics = new MetricsProcessor(layouts, labels, namespace, binding);
-            do
-            {
-                metrics.print(System.out);
-                Thread.sleep(interval * 1000L);
-            } while (interval != 0);
-        }
-        catch (InterruptedException | IOException ex)
+            printer.print(System.out);
+            sleep(interval);
+        } while (interval != 0);
+        close(engine);
+    }
+
+    private Engine engine()
+    {
+        final Configuration config = engineConfiguration();
+        final ErrorHandler onError = Throwable::printStackTrace;
+
+        try (Engine engine = Engine.builder()
+            .config(config)
+            .errorHandler(onError)
+            .readOnly()
+            .build())
         {
-            LangUtil.rethrowUnchecked(ex);
+            engine.start();
+            return engine;
```

Comment:
Feels wrong to implicitly close the engine and then keep using it after it has been closed.

Perhaps it would be better to put this `try-with-resources` block around the executable behavior of the metrics command sleep loop instead

Reply:
The `engine` is explicitly closed as the last command in the `run` method here:
https://github.com/attilakreiner/zilla/blob/2904423083647c43ba079eb0b2a23d5e0b823dbe/runtime/command-metrics/src/main/java/io/aklivity/zilla/runtime/command/metrics/internal/airline/ZillaMetricsCommand.java#L79
https://github.com/attilakreiner/zilla/blob/2904423083647c43ba079eb0b2a23d5e0b823dbe/runtime/command-metrics/src/main/java/io/aklivity/zilla/runtime/command/metrics/internal/airline/ZillaMetricsCommand.java#L125-L136

Reply:
Yes, understood - had already seen that when making the comment above, it doesn't address the feedback, that we are letting the engine escape from try-with-resources (which already closes it implicitly at the end of the try-with-resources block), then return it and use it from run method (when already closed), then attempt to explicitly close it (again) at the end of the run method.

Reply:
yep, got it; done